src/main/java/reposense/authorship/analyzer/AnnotatorAnalyzer.java

```diff
@@ -22,12 +23,20 @@
  */
 public class AnnotatorAnalyzer {
     private static final String AUTHOR_TAG = "@@author";
-    // reference: https://github.com/shinnn/github-username-regex
+    // GitHub username format
     private static final String REGEX_AUTHOR_NAME_FORMAT = "^[a-zA-Z0-9](?:[a-zA-Z0-9]|-(?=[a-zA-Z0-9])){0,38}$";
     private static final Pattern PATTERN_AUTHOR_NAME_FORMAT = Pattern.compile(REGEX_AUTHOR_NAME_FORMAT);
-    private static final String REGEX_COMMENT_WITH_AUTHOR_TAG_FORMAT = "^[^\\w]*@@author(\\s[^\\s]+)?[\\s]*$";
-    private static final Pattern PATTERN_COMMENT_WITH_AUTHOR_TAG_FORMAT =
-            Pattern.compile(REGEX_COMMENT_WITH_AUTHOR_TAG_FORMAT);
+    private static final String REGEX_AUTHOR_TAG_FORMAT = "@@author(\\s[^\\s]+)?";
+
+    private static final String[][] COMMENT_FORMATS = {
+            {"//", "\\s"},
+            {"/\\*", "\\*/"},
+            {"#", "\\s"},
+            {"<!--", "-->"},
+            {"%", "\\s"},
+    };
+    private static ArrayList<Pattern> commentPatterns = new ArrayList<>();
+    private static boolean patternsGenerated = false;
```

Comment:
Having these two lines makes the design seem a little weird. It does not seem to make the functions truly independent in terms of state. I don't have any concrete suggestions, but is it possible to implement this slightly differently so we can follow OOP principles?

Reply:
ok!